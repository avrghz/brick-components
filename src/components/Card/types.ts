export const SHADOW = ['never', 'always', 'hover'] as const

export type Shadow = typeof SHADOW[number]
