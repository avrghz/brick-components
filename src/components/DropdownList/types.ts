export interface Option {
    label: string
    value: string
    disabled?: boolean
}

export type Callback = (index: number) => void
