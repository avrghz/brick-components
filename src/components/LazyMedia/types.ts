export type MediaType = 'img' | 'picture'
