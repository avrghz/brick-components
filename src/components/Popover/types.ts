export const TRIGGER_ON = ['click', 'focus', 'hover', 'manual'] as const

export type TriggerOn = typeof TRIGGER_ON[number]
