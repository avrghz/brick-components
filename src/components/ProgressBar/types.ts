export const BAR_TYPE = ['line', 'circle', 'dashboard'] as const

export type BarType = typeof BAR_TYPE[number]

export interface ProgressProps {
    progress: number
    showPercentageText?: boolean
    textInside?: boolean
}

export interface ProgressTextProps extends Pick<ProgressProps, 'progress'> {
    show?: boolean
    isInnerText?: boolean
}

export interface ProgressCircleProps extends ProgressProps {
    strokeDash: [number, number]
    strokeDashOffset: number
    flip?: boolean
}
