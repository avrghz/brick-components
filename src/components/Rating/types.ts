export interface StarProps {
    hover?: boolean
    activeColor?: string
}

export interface Colors {
    [key: number]: string
}
