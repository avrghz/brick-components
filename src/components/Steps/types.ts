import { JSXBase } from '@stencil/core/internal'

export const STEP_STATE = ['process', 'wait', 'success', 'error', 'finish'] as const

export interface StepComponent {
    step: number
    title: string
    description?: string
    state?: typeof STEP_STATE[number]
}

export interface StepProps extends StepComponent {
    style?: JSXBase.HTMLAttributes<HTMLDivElement>['style']
    isLast?: boolean
    isCentered?: boolean
}
