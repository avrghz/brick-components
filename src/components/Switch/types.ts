export interface AVChange {
    isOn: boolean
}
