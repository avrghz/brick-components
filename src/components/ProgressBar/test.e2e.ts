import { newE2EPage, E2EPage, E2EElement } from '@stencil/core/testing'
import { VARIANTS } from '../../shared/types'
import { BAR_TYPE } from './types'

describe('<bk-progress-bar/>', () => {
    let page: E2EPage
    let component: E2EElement

    const getProgressBar = async () => await page.find('bk-progress-bar >>> .bk-progress')

    const getProgressText = async () => await page.find('bk-progress-bar >>> .bk-progress__text')

    beforeEach(async () => {
        page = await newE2EPage()
        await page.setContent(`<bk-progress-bar progress="75"/>`)
        component = await page.find('bk-progress-bar')
    })

    it('should render', async () => {
        expect(component).not.toBeNull()
    })

    describe('Prop: progress', () => {
        it(`it should render with given progress percentage`, async () => {
            component.setProperty('progress', 10)
            await page.waitForChanges()
            expect(await getProgressBar()).toMatchSnapshot('prop_progress')
        })
    })

    describe('Prop: showPercentageText', () => {
        BAR_TYPE.forEach((type) => {
            it(`should not show percentage text, when set to false for ${type} bar`, async () => {
                component.setProperty('variant', type)
                component.setProperty('showPercentageText', false)
                await page.waitForChanges()
                expect(await getProgressText()).toBeNull()
            })

            it(`should show percentage text, when set to true for ${type} bar`, async () => {
                component.setProperty('variant', type)
                component.setProperty('showPercentageText', true)
                await page.waitForChanges()
                expect(await getProgressText()).not.toBeNull()
            })
        })
    })

    describe('Prop: textInside', () => {
        it('should render with percentage inside the bar, when set to true', async () => {
            component.setProperty('textInside', true)
            await page.waitForChanges()
            const innerText = await (await (await getProgressBar()).find('.bk-progress-bar')).find('.bk-progress__text')
            expect(innerText).not.toBeNull()
        })

        it('should render with percentage outside the bar, when set to false', async () => {
            component.setProperty('textInside', false)
            await page.waitForChanges()
            const innerText = await (await (await getProgressBar()).find('.bk-progress-bar')).find('.bk-progress__text')
            const outerText = await getProgressText()
            expect(outerText).not.toBeNull()
            expect(innerText).toBeNull()
        })
    })

    describe('Prop: variant', () => {
        VARIANTS.forEach((variant) => {
            it(`it should render as ${variant}`, async () => {
                component.setProperty('variant', variant)
                await page.waitForChanges()
                expect(await getProgressBar()).toHaveClass(variant)
            })
        })
    })

    describe('Prop: barType', () => {
        BAR_TYPE.forEach((type) => {
            it(`it should render as ${type}`, async () => {
                component.setProperty('barType', type)
                await page.waitForChanges()
                expect(await getProgressBar()).toMatchSnapshot('prop_bar_type')
            })
        })
    })
})
