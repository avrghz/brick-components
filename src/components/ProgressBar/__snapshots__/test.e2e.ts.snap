// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bk-progress-bar/> Prop: barType it should render as circle: prop_bar_type 1`] = `
<div class="bk-progress default">
  <div class="bk-progress-circle">
    <style>
      .bk-progress-circle__inner{
                    stroke-dasharray: 221.48250000000002 295.31;
                    stroke-dashoffset: 0;
                }

                .bk-progress-circle__outer{
                    stroke-dasharray: 295.31 295.31;
                    stroke-dashoffset: 0;
                }
    </style>
    <svg viewBox="0 0 100 100">
      <path class="bk-progress-circle__outer" d="
        M 50 50
        m 0 -47
        a 47 47 0 1 1 0 94
        a 47 47 0 1 1 0 -94
        " fill="none"></path>
      <path class="bk-progress-circle__inner" d="
        M 50 50
        m 0 -47
        a 47 47 0 1 1 0 94
        a 47 47 0 1 1 0 -94
        " fill="none"></path>
    </svg>
    <div class="bk-progress__text">
      75%
    </div>
  </div>
</div>
`;

exports[`<bk-progress-bar/> Prop: barType it should render as dashboard: prop_bar_type 1`] = `
<div class="bk-progress default">
  <div class="bk-progress-circle">
    <style>
      .bk-progress-circle__inner{
                    stroke-dasharray: 166.1115 295.31;
                    stroke-dashoffset: -36.9137;
                }

                .bk-progress-circle__outer{
                    stroke-dasharray: 221.482 295.31;
                    stroke-dashoffset: -36.9137;
                }
    </style>
    <svg class="bk-progress--flip" viewBox="0 0 100 100">
      <path class="bk-progress-circle__outer" d="
        M 50 50
        m 0 -47
        a 47 47 0 1 1 0 94
        a 47 47 0 1 1 0 -94
        " fill="none"></path>
      <path class="bk-progress-circle__inner" d="
        M 50 50
        m 0 -47
        a 47 47 0 1 1 0 94
        a 47 47 0 1 1 0 -94
        " fill="none"></path>
    </svg>
    <div class="bk-progress__text">
      75%
    </div>
  </div>
</div>
`;

exports[`<bk-progress-bar/> Prop: barType it should render as line: prop_bar_type 1`] = `
<div class="bk-progress default">
  <div class="bk-progress-bar">
    <div class="bk-progress-bar__outer">
      <div class="bk-progress-bar__inner" style="width: 75%;"></div>
    </div>
  </div>
  <div class="bk-progress__text">
    75%
  </div>
</div>
`;

exports[`<bk-progress-bar/> Prop: progress it should render with given progress percentage: prop_progress 1`] = `
<div class="bk-progress default">
  <div class="bk-progress-bar">
    <div class="bk-progress-bar__outer">
      <div class="bk-progress-bar__inner" style="width: 10%;"></div>
    </div>
  </div>
  <div class="bk-progress__text">
    10%
  </div>
</div>
`;
